internal/build.go

```diff
@@ -81,9 +79,17 @@ func Build(ctx context.Context, opts BuildOptions) (*gobl.Envelope, error) {
 	if err != nil {
 		return nil, err
 	}
-	env := new(gobl.Envelope)
-	if err = json.Unmarshal(encoded, env); err != nil {
-		return nil, echo.NewHTTPError(http.StatusBadRequest, err.Error())
+	var env *gobl.Envelope
+	if opts.Envelop {
```

Comment:
The fact that we're checking this same flag in several places makes me nervous. This pattern makes for tightly coupled code, and is very error prone.

I'd much rather have a `Build()` and `Envelop()` method respectively, that then call unexported methods for shared functionality.

Reply:
My feeling was that you'd end up with loads of duplicated code. If you'd like to refactor this so it fits your proposed style, please do so! I was most interested in a quick fix to test out the "feel" of the "envelop" command. I accomplished that, and still like it, so please feel free to make it better :-) I'm current focussed on conversations and documentation, so it'll take me a bit more effort to come back to this!

Reply:
I've given it a go. It's a bit better, although I'm not 100% convinced :-)